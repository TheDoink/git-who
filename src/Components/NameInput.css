#nameInputForm {
	margin-top:10px;
	width:100%;
}

#nameInputElement {
	vertical-align: bottom;
	margin:0;
	padding: 0;
}

/* The actual input element */
#nameInput {
	width:90%;

	border:none;
	
	height:10vw;
	padding:0px;
	font-size:8vw;

	border-radius:2vw 0vw 0vw 2vw;
	-moz-border-radius:2vw 0vw 0vw 2vw;
}

/* The button for submitting */
#submitButton {

	background:#078800;
	display: inline-block;
	font-size:8vw;
	height:10vw;
	width:10%;
	text-align: center;
	vertical-align: top;
	cursor:pointer;

	border-radius:0vw 2vw 2vw 0vw;
	-moz-border-radius:0vw 2vw 2vw 0vw;
}