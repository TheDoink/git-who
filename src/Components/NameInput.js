import React, { Component } from 'react';
import './NameInput.css';
import $ from 'jquery';

class NameInput extends Component {

  constructor(props) {
    super(props);

    this.state = {
      name: ""
    }
  }

  /* Validation for users typing in names */
  handleUpdate = (e) => {
    e.preventDefault();

    let val = e.target.value;

    /* Github has a maxLength of usernames for 38 chars */
    if(val.length <= 38) {
      this.setState({name: val}, function() {
        console.log(this.state);
      });
    }
  };

  handleSubmit = (e) => {

    // Prevent the usual behavior
    e.preventDefault();

    let name = this.state.name;

    /* Github regex found from: https://github.com/shinnn/github-username-regex
       Regex was also validated by github username requirements page */
    let githubUsernameRegex = /^[a-z\d](?:[a-z\d]|-(?=[a-z\d])){0,38}$/i;

    // If we pass the regex...
    if(githubUsernameRegex.test(name)) {

      // Try and get that username...
      $.get(`https://api.github.com/users/${name}`,

        // If we get a good response...
        (success, text, obj) => {

          console.log(success);

          // This sets the user and propogates it up
          this.props.setUser({status: obj.status, data: success});
          this.setState({name: ""}, function() {});
        })

        // If we get anything else...
        .fail((failure, text, error) => {
          this.props.setUser({status: failure.status, data:{}});
        });
      
    }
  };

  render() {
    
    return (
      <form id="nameInputForm" onSubmit={this.handleSubmit.bind(this)}>
        <div id="nameInputElement">
          <input id="nameInput" placeholder="Enter A Git Username" onChange={this.handleUpdate.bind(this)} type="text" value={this.state.name}/>
          <span id="submitButton" onClick={this.handleSubmit.bind(this)}>
            <i className="fa fa-arrow-circle-o-right"></i>
          </span>
        </div>
      </form>
    );
  }
}

NameInput.propTypes = {
  name: React.PropTypes.string
}

export default NameInput;
