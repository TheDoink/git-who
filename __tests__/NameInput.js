import React from 'react';
import renderer from 'react-test-renderer';
import NameInput from '../src/Components/NameInput';

it('lines up with the snapshot', () => {
  let testMe = renderer.create(<NameInput setUser={function(){}}></NameInput>).toJSON();

  expect(testMe).toMatchSnapshot();
  // testMe.getGist('d2eace7977e2aeaf942de0f8b3e7275c');

  
});
