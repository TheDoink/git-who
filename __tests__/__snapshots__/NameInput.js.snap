// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lines up with the snapshot 1`] = `
<form
  id="nameInputForm"
  onSubmit={[Function]}
>
  <div
    id="nameInputElement"
  >
    <input
      id="nameInput"
      onChange={[Function]}
      placeholder="Enter A Git Username"
      type="text"
      value=""
    />
    <span
      id="submitButton"
      onClick={[Function]}
    >
      <i
        className="fa fa-arrow-circle-o-right"
      />
    </span>
  </div>
</form>
`;
